src/components/founderDetail/OverviewBoxButtons.tsx

```diff
@@ -10,37 +10,59 @@ import { Button } from "@tremor/react";
 import { useState } from "react";
 import { remult } from "remult";
 import LinkButton from "../buttons/LinkButton";
+import { SchoolTypeValues, getSchoolTypeCode } from "@/entities/School";
+import { getStatusPropertyBySchoolType } from "@/entities/City";
 
 export default function OverviewBoxButtons({
   city,
   fetchCity,
   activeOrdinanceId,
   urlFrom,
+  rootPath,
+  schoolType,
 }: {
   city: City;
   fetchCity: () => Promise<void>;
   activeOrdinanceId?: number;
   urlFrom?: string[];
+  rootPath: string;
+  schoolType: string;
 }) {
   const [loading, setLoading] = useState<boolean>(false);
 
+  const schoolTypeCode = getSchoolTypeCode(schoolType);
+  const statusParam = getStatusPropertyBySchoolType(schoolTypeCode);
+
   const setAsPublished = async () => {
     setLoading(true);
-    await remult.repo(City).save({ ...city, statusElementary: CityStatus.Published });
+
+    let statusObject: {
+      [key: string]: CityStatus;
+    } = {};
+
+    statusObject[statusParam] = CityStatus.Published;
+
+    await remult.repo(City).save({ ...city, ...statusObject });
     await fetchCity();
     setLoading(false);
   };
 
   const setAsInProgress = async () => {
     setLoading(true);
-    await remult.repo(City).save({ ...city, statusElementary: CityStatus.InProgress });
+    let statusObject: {
+      [key: string]: CityStatus;
+    } = {};
+
+    statusObject[statusParam] = CityStatus.InProgress;
+
+    await remult.repo(City).save({ ...city, ...statusObject });

```

Comment:
You could simplify this like this:
```ts
await remult.repo(City).save({ ...city, [statusParam]: CityStatus.Published });
```

This would give you a type error, as the return type of `getStatusPropertyBySchoolType` is string, so you need to either remove the explicit return type or change it to `"statusElementary" | "statusKindergarten"`.

The same applies to the `setAsPublished` function above.